### PR TITLE
BDT Example and readme update

### PR DESCRIPTION
**Pull request by justin-wong-ce**
Added BroadcastDistributionTable example and updated version number

### CODE REVIEW

File: BACnetServerExample/BACnetServerExample.cpp [hunk 1]
```diff
@@ -566,6 +566,13 @@ int main(int argc, char** argv)
 		std::cerr << "Failed to add NetworkPort" << std::endl;
 		return -1;
 	}
+	fpSetPropertyEnabled(g_database.device.instance, CASBACnetStackExampleConstants::OBJECT_TYPE_NETWORK_PORT, g_database.networkPort.instance, CASBACnetStackExampleConstants::PROPERTY_IDENTIFIER_BBMD_ACCEPT_FD_REGISTRATIONS, true);
+	fpSetPropertyEnabled(g_database.device.instance, CASBACnetStackExampleConstants::OBJECT_TYPE_NETWORK_PORT, g_database.networkPort.instance, CASBACnetStackExampleConstants::PROPERTY_IDENTIFIER_BBMD_BROADCAST_DISTRIBUTION_TABLE, true);
+	fpSetPropertyEnabled(g_database.device.instance, CASBACnetStackExampleConstants::OBJECT_TYPE_NETWORK_PORT, g_database.networkPort.instance, CASBACnetStackExampleConstants::PROPERTY_IDENTIFIER_BBMD_FOREIGN_DEVICE_TABLE, true);
+	uint8_t bbmdIpAddress[6] = {192, 168, 1, 208, 0xBA, 0xC0};
```
Comment (ssmethurst) [inline BACnetServerExample/BACnetServerExample.cpp:572]: Can't used hard-coded variables in the example

**Comment by justin-wong-ce** [comment]
Added debug callback example
Added comment to notify users about CAS BACnet Stack
Removed hardcoded BBMD Address/Port/Mask, instead asks user for input

**Comment by justin-wong-ce** [comment]
Bumped to version 0.0.14

File: BACnetServerExample/BACnetServerExample.cpp [hunk 2]
```diff
@@ -2075,6 +2129,16 @@ bool CallbackDeviceCommunicationControl(const uint32_t deviceInstance, const uin
 	return true;
 }
 
+void CallbackLogDebugMessage(const char* message, const uint16_t messageLength, const uint8_t messageType) {
+	// This callback is called when the CAS BACnet Stack logs an error or info message
```
Comment (ssmethurst) [inline BACnetServerExample/BACnetServerExample.cpp:2133]: Just dump the string to stdout in the call-back function. No need to make a global function

File: BACnetServerExample/BACnetServerExample.cpp [hunk 3]
```diff
@@ -610,6 +657,13 @@ int main(int argc, char** argv)
 		// Update values in the example database
 		g_database.Loop();
 
+		// Check for debug messages
+		if (debugMessage.c_str() != g_database.debugMessage && g_database.debugMessage != "") {
```
Comment (ssmethurst) [inline BACnetServerExample/BACnetServerExample.cpp:661]: Remove this section, and just do this work in the callback. 

Also just dump the message. No need to prefix it with anything.

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,12 @@
 
 ## Version 0.0.x
 
+### 0.0.14.x (2021-Jan-20)
+- Added LogDebugMessage callback to example
```
Comment (ssmethurst) [inline CHANGELOG.md:6]: Mention the new version of the CAS BACnet Stack that has been used with the LogDebugMessage

- Updated CAS BACnet Stack to version XXXXX

File: BACnetServerExample/BACnetServerExample.cpp [hunk 5]
```diff
@@ -566,7 +574,45 @@ int main(int argc, char** argv)
 		std::cerr << "Failed to add NetworkPort" << std::endl;
 		return -1;
 	}
-	std::cout << "OK" << std::endl;
+	fpSetPropertyEnabled(g_database.device.instance, CASBACnetStackExampleConstants::OBJECT_TYPE_NETWORK_PORT, g_database.networkPort.instance, CASBACnetStackExampleConstants::PROPERTY_IDENTIFIER_BBMD_ACCEPT_FD_REGISTRATIONS, true);
+	fpSetPropertyEnabled(g_database.device.instance, CASBACnetStackExampleConstants::OBJECT_TYPE_NETWORK_PORT, g_database.networkPort.instance, CASBACnetStackExampleConstants::PROPERTY_IDENTIFIER_BBMD_BROADCAST_DISTRIBUTION_TABLE, true);
+	fpSetPropertyEnabled(g_database.device.instance, CASBACnetStackExampleConstants::OBJECT_TYPE_NETWORK_PORT, g_database.networkPort.instance, CASBACnetStackExampleConstants::PROPERTY_IDENTIFIER_BBMD_FOREIGN_DEVICE_TABLE, true);
+	
+	// Ask for BBMD Address, Port, and Mask
```
Comment (ssmethurst) [inline BACnetServerExample/BACnetServerExample.cpp:581]: This shouldn't be asked on start up as most people won't have a BBMD to connect to. Instead make this part of the UserInput function

**Pull request merged**